### ISSUE TITLE
Object Explorer Connect/Expand Action Takes ~35 Seconds on Mac (Integrated Auth)

### ISSUE THREAD

**Issue by chlafreniere** [microsoft/azuredatastudio#22125]
(latest insiders, amd64 Mac, kerberos auth over VPN, connecting to SQL server 2022 instance on prem)

Took ~35 seconds from when I clicked on a server from OE until expansion was completed (i.e. until I could see the "Databases" folder). Captured a sqltools.log file, which I'll share offline (as it contains instance names).

Was able to repro very similar behavior with an RC1 build from a previous release, so this doesn't appear to be new behavior.

A couple of quick notes:
- I noticed some behavior that, to an untrained eye, looks suspiciously similar to a couple of 5 second timeouts in the log:
```
23-03-02 20:09:51.1685500 pid:9188 tid:14 sqltools Verbose: 0 : eventTID:1769616 TdsParser.Connect | SEC | SSPI or Active Directory Authentication Library loaded for SQL Server based integrated authentication
23-03-02 20:09:56.2788020 pid:9188 tid:14 sqltools Verbose: 0 : eventTID:1769616 SNIProxy.GetSqlServerSPN | Info | ServerName <<REDACTED>>, InstanceName null, Port -1, postfix null
23-03-02 20:10:01.2826480 pid:9188 tid:14 sqltools Verbose: 0 : eventTID:1769616 SNIProxy.CreateConnectionHandle | Info | Session Id 67795fb5-ef04-42a8-b7da-24ca503b3647, SNI Handle Type: Microsoft.Data.SqlClient.SNI.SNITCPHandle
```
- It appears that there are quite a number of queries that are being executed as part of connecting/OE expansion. Are they really all needed for just initial expansion (see log files for details)?

(Unsure if I should label this Object Explorer or Connection, but I went with Object Explorer. Apologies if that's not the right label in this case)

**Comment by cheenamalhotra** [comment]
Update from internal investigations:
The time taken here is by the local Active Directory in resolving IP address for provided server name as is also reproduced with calls to below API:

1. `Dns.GetHostAddresses(serverName)` (5 seconds)
2. `Dns.GetHostEntry(hostNameOrAddress)` (5 seconds)

Also captured with SqlClient logs:
```
2023-03-31T16:10:32.8958200-07:00 SNICommon.GetDnsIpAddresses | INFO | Getting DNS host entries for serverName sqltools2022-3.
2023-03-31T16:10:37.9032020-07:00 Exit SNI Scope 2
2023-03-31T16:10:37.9169690-07:00 SNITCPHandle.Connect | INFO | Connecting to IP address 10.116.97.135 and port 1433 using InterNetwork address family.
2023-03-31T16:10:37.9971260-07:00 SNITCPHandle..ctor | INFO | Connection Id b6a42f59-9cad-4570-bbcb-c92578866ab0 Socket opened successfully, TCP stream ready.
2023-03-31T16:10:37.9975160-07:00 Exit SNI Scope 1
2023-03-31T16:10:37.9983000-07:00 SNIProxy.GetSqlServerSPN | Info | ServerName sqltools2022-3, InstanceName null, Port -1, postfix null
2023-03-31T16:10:42.9996370-07:00 SNIProxy.GetSqlServerSPN | Info | ServerSPNs MSSQLSvc/sqltools2022-3.redmond.corp.microsoft.com and MSSQLSvc/sqltools2022-3.redmond.corp.microsoft.com:1433
```

And since there are 3 connections made during login, it amounts to 30 seconds+ in this case.

This is likely an issue from [Dns](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs) APIs, will continue to investigate any opportunities to improve performance.

**Comment by cheenamalhotra** [comment]
Hi @chlafreniere 

Can you please try with the latest insiders build, as we've enabled connection pooling by default for next release - that should improve connection performance when experiencing DNS delays.

**Comment by cheenamalhotra** [comment]
Re-iterating, issue can be addressed by enabling connection pooling setting - now available in 1.45.1 and enabled by default in Insiders. Also with latest changes, we've moved the metadata queries to run in non-blocking way so even less time will be taken to complete connection. I'll proceed to close the issue as resolved and DNS issues are external to ADS/STS.
